### PR TITLE
Fix vulnerabilities in dependencies

### PR DESCRIPTION
**Pull request by halfpastfouram**
The outdated and vulnerable esdoc2 prevented dependency updates so I've replaced it with jsdoc which works practically the same way. 

After this PR I'd like to focus on adding more methods like checking on and updating store inventories.

### CODE REVIEW

File: src/oAuthHelper.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import * as crypto from 'crypto';
 import oauthSignature from 'oauth-signature';
+import randomBytes from 'randombytes';
```
Comment (ryansh100) [inline src/oAuthHelper.js:2]: Are you trying to make this library work from the browser?

I would have some concerns since the Bricklink API requires a client secret that should not be published to a browser.

If not, we should just use the Node runtime's crypto library.

File: src/client.js [hunk 2]
```diff
@@ -49,7 +48,7 @@ export class Client {
    */
   send(req) {
     const promise = new Promise((resolve, reject) => {
-      const callback = () => this.dispatch(req).then(resolve).catch(reject).finally(() => {
```
Comment (ryansh100) [inline src/client.js:52]: What is the reason for this change?

File: src/client.js [hunk 3]
```diff
@@ -8,7 +8,6 @@ import { Superset } from './catalogItem/supersets';
 import { logger } from './logger';
 import { BricklinkRequest } from './request';
 import { BrickLinkApiError } from './brickLinkApiError';
-import fetch from 'node-fetch'
```
Comment (ryansh100) [inline src/client.js:11]: praise: Glad to have this reference removed.

File: package.json [hunk 4]
```diff
@@ -33,31 +33,31 @@
...
+    "pretty-quick": "^3.1.3",
+    "typescript": "^5.1.6"
   },
   "dependencies": {
-    "debug": "^4.3.2",
-    "node-fetch": "^2.6.7",
-    "oauth-signature": "^1.3.1"
+    "crypto": "^1.0.1",
+    "http": "^0.0.1-security",
+    "jsdoc": "^4.0.2",
```
Comment (ryansh100) [inline package.json:58]: move to devDependencies please.

File: package.json [hunk 5]
```diff
@@ -33,31 +33,31 @@
...
+    "prettier": "^3.0.0",
+    "pretty-quick": "^3.1.3",
+    "typescript": "^5.1.6"
   },
   "dependencies": {
-    "debug": "^4.3.2",
-    "node-fetch": "^2.6.7",
-    "oauth-signature": "^1.3.1"
+    "crypto": "^1.0.1",
+    "http": "^0.0.1-security",
```
Comment (ryansh100) [inline package.json:57]: I'm not sure why we would need crypto or http npm packages as dependencies. Please remove references if possible.

File: tsconfig.json [hunk 6]
```diff
@@ -10,4 +12,8 @@
     "include": [
         "src/**/*"
     ],
-}
\ No newline at end of file
+    "exclude": [
+        "/dist",
+        "/node_modules"
+    ]
```
Comment (ryansh100) [inline tsconfig.json:18]: praise: thank you again for this clean up

**Review by ryansh100 — CHANGES_REQUESTED** [review]
Appreciate someone updating packages and fixing some issues.

I would appreciate some further comments on suggested changes.

File: src/client.js (hunk 2 above)
Comment (halfpastfouram) [reply src/client.js:52]: Moved from ES5 to ES6. `.finally` is not available on promises:

```
error TS2550: Property 'finally' does not exist on type 'Promise<void>'
```

File: src/oAuthHelper.js (hunk 1 above)
Comment (ryansh100) [reply src/oAuthHelper.js:2]: randomBytes is not deprecated. Maybe the import should be changed to import 'node:crypto' if there is an npm package showing up in the tool chain.
https://nodejs.org/api/crypto.html#cryptorandombytessize-callback

Comment (halfpastfouram) [reply src/oAuthHelper.js:2]: Right. I (or node) couldn't find the `randomBytes` method but I think now I'll just have to install `@types/node` as dev dependency. I'll try that later tonight.

File: package.json (hunk 5 above)
Comment (halfpastfouram) [reply package.json]: Done.

File: package.json (hunk 4 above)
Comment (halfpastfouram) [reply package.json]: Done.

**Review by ryansh100 — APPROVED** [review]

**Comment by ryansh100** [comment]
There were some things that needed to be brought back to continue Node 12 support.

There's an argument for a breaking change that brings package up to Node >16, but that's a different PR.

**Comment by ryansh100** [comment]
I did merge a branch that resolves the audit issues and published a new version.
